onadata/apps/api/tests/viewsets/test_data_viewset.py

```diff
@@ -1591,6 +1592,61 @@ def test_delete_submission_inactive_form(self, send_message_mock):
         self.assertEqual(response.status_code, 400)
         self.assertTrue(send_message_mock.called)
 
+    @patch('onadata.apps.api.viewsets.data_viewset.send_message')
+    def test_delete_submission_subset(self, send_message_mock):
+        xls_file_path = os.path.join(
+            os.path.dirname(os.path.abspath(__file__)),
+            "../fixtures/tutorial/tutorial.xls"
+        )
+        self._publish_xls_file_and_set_xform(xls_file_path)
+        # Add multiple submissions
+        for x in range(1, 11):
+            path = os.path.join(
+                settings.PROJECT_ROOT, 'libs', 'tests', "utils", 'fixtures',
+                'tutorial', 'instances', 'uuid{}'.format(x), 'submission.xml')
+            self._make_submission(path)
+            x += 1
+        self.xform.refresh_from_db()
+        formid = self.xform.pk
+        initial_count = self.xform.instances.filter(deleted_at=None).count()
+        self.assertEqual(initial_count, 9)
+        self.assertEqual(self.xform.num_of_submissions, 9)
+
+        view = DataViewSet.as_view({'delete': 'destroy'})
+        deleted_instances_subset = self.xform.instances.all()[:6]
+        instance_ids = ','.join([str(i.pk) for i in deleted_instances_subset])
+        data = {"instance_ids": instance_ids, 'delete_all': False}
+
+        request = self.factory.delete('/', data=data, **self.extra)
+        response = view(request, pk=formid)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            response.data.get('message'),
+            "%d records were deleted" % len(deleted_instances_subset)
+        )
+        self.assertTrue(send_message_mock.called)
+        send_message_mock.called_with(
+            [str(i.pk) for i in deleted_instances_subset], formid, XFORM,
+            request.user, SUBMISSION_DELETED)
+
+        # Test that num of submissions for the form is successfully updated
+        self.xform.refresh_from_db()
+        current_count = self.xform.instances.filter(deleted_at=None).count()
+        self.assertNotEqual(current_count, initial_count)
+        self.assertEqual(current_count, 3)
+        self.assertEqual(self.xform.num_of_submissions, 3)
+
+        # Test delete_all submissions for the form
+        data = {'delete_all': True}
+        request = self.factory.delete('/', data=data, **self.extra)
+        response = view(request, pk=formid)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            response.data.get('message'),
+            "3 records were deleted")
```

Comment:
Could we also confirm that all the post delete actions are run accordingly i.e XForm submission count is updated on the API as well as the Project date_modified e.t.c

Reply:
A check to confirm that the xform submission count is updated can be found on lines 1633 - 1638